docs/snippets/solid/component-story-custom-args-complex.js.mdx

```diff
@@ -0,0 +1,47 @@
+```js
+// YourComponent.stories.js|jsx
+
+import { createSignal, createEffect } from 'solid-js';
+import { YourComponent } from './your-component';
+
+export default {
+  /* 👇 The title prop is optional.
+   * See https://storybook.js.org/docs/7.0/solid/configure/overview#configure-story-loading
+   * to learn how to generate automatic titles
+   */
+  title: 'YourComponent',
+  component: YourComponent,
+  //👇 Creates specific argTypes with options
+  argTypes: {
+    propertyA: {
+      options: ['Item One', 'Item Two', 'Item Three'],
+      control: { type: 'select' }, // Automatically inferred when 'options' is defined
+    },
+    propertyB: {
+      options: ['Another Item One', 'Another Item Two', 'Another Item Three'],
+    },
+  },
+};
+
+//👇 Some function to demonstrate the behavior
+const someFunction = (valuePropertyA, valuePropertyB) => {
+  // Makes some computations and returns something
+};
+
+export const ExampleStory = {
+  render: (args) => {
+    const [someFunctionResult, setSomeFunctionResult] = createSignal();
```

Comment:
I'm not very familiar with SolidJS. Is there someone from the community that could review the SolidJS-specific parts of this work?

Reply:
I can do the review of the code snippets on a separate sandbox project, so I can attach a link to this PR with the sample repo so you can test from those implementations. Let me know if this works for you.